### PR TITLE
Attempt to add a PReP partition.

### PR DESCRIPTION
**Pull request by xnox**
Bootloader fails to install on POWER because there is no prep partition.

Ensure we create one.

Not sure how this works at all, in neither subiquity or curtin. But it seems like curtin should know what to do with a 'prep' flag partition. I guess it's worth a try.

### CODE REVIEW

File: subiquity/ui/views/filesystem/partition.py [hunk 1]
```diff
@@ -208,6 +208,13 @@ def validate_mount(self):
     'installed onto this partition, which must be formatted as fat32. The '
     'only aspect of this partition that can be edited is the size.')
 
+prep_partition_description = _(
+    "Required bootloader partition\n"
+    "\n"
+    'This is the PReP partion which is required on POWER systems. Grub will be '
+    'installed onto this partition. The '
+    'only aspect of this partition that can be edited is the size.')
```
Comment (raharper) [inline subiquity/ui/views/filesystem/partition.py]: I don't think the size should be editable; or at least we should require a minimum size.  I've seen recommendations for 8, 10, 16 and 32MB in size.

**Comment by raharper** [comment]
Curtin does indeed know what to do with prep flag, curtin will find the prep partition and pass that as the device to install grub.

File: subiquity/ui/views/filesystem/partition.py (hunk 1 above)
Comment (xnox) [reply subiquity/ui/views/filesystem/partition.py]: Yeah, I have no idea what I am doing here. @adconrad is PReP partition always 8MiB? Fixed size? Allow users to twiddle it or not?

**Comment by raharper** [comment]
> This looks basically fine. Are we going to need to add anything like this for s390x?

No special partitions are needed for zipl.

File: subiquity/models/filesystem.py [hunk 2]
```diff
@@ -483,7 +483,7 @@ def short_label(self):
         return _("partition {}").format(self._number)
 
     def available(self):
-        if self.flag == 'bios_grub':
+        if self.flag in ['bios_grub', 'prep']:
```
Comment (zyga) [inline subiquity/models/filesystem.py]:
```suggestion
        if self.flag in ('bios_grub', 'prep'):
```

**Comment by xnox** [comment]
> This looks basically fine. Are we going to need to add anything like this for s390x?

s390x is already done a while back, by removing the "make bootable" action / requirements.

**Comment by mwhudson** [comment]
Oh didn't see the comment from @zyga , still that's a minor thing.

**Pull request merged**